samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <uart.h>
+#include <string.h>
+#include <at_host.h>
+
+#include <net/mqtt.h>
+#include <net/socket.h>
+#include <lte_lc.h>
+
+#define APP_MQTT_BUFFER_SIZE 128
+
+/* Buffers for MQTT client. */
+static u8_t rx_buffer[APP_MQTT_BUFFER_SIZE];
+static u8_t tx_buffer[APP_MQTT_BUFFER_SIZE];
+
+/* The mqtt client struct */
+static struct mqtt_client client;
+
+/* MQTT Broker details. */
+static struct sockaddr_storage broker;
+
+static bool connected;
+
+static void data_print(u8_t *prefix, u8_t *data, size_t len)
```

Comment:
Please add a comment to mention that this function is necessary because some strings are not `NULL` terminated and can't be printed with `printk` directly.